docs/use-cases/microfrontends.md

```diff
@@ -0,0 +1,308 @@
+└── staging
+│   ├── datafile-tag-products.json
+│   ├── datafile-tag-signup.json
+│   ├── datafile-tag-signin.json
+│   ├── datafile-tag-checkout.json
+│   └── datafile-tag-account.json
+
+2 directories, 10 files
+```
+
+## Attributes
+
+We will set up some Featurevisor attributes first, that we will use in various stages of this guide later.
+
+### `userId`
+
+```yml
+# attributes/userId.yml
+description: Unique identifier of the logged in User
+type: string
+capture: true
+```
+
+### `deviceId`
+
+This ID can be generated at the client-side level when user first visits your app, and stored in a cookie or localStorage for e.g. if in a browser environment.
+
+```yml
+# attributes/deviceId.yml
+description: Unique identifier of the device
+type: string
+capture: true
+```
+
+## Feature
+
+For this example, we will create a new feature flag that's responsible for toggling a marketing banner that's shown at the bottom of a page. We can call it `showMarketingBanner`.
+
+```yml
+# features/showMarketingBanner.yml
+description: Shows marketing banner at the bottom of the page
+tags:
+  - <your-tag-here> # we will discuss this in next section below
+
+defaultVariation: false
+
+bucketBy: deviceId
+
+variations:
+  - value: true
+    weight: 100
+
+  - value: false
+    weight: 0
+
+environments:
+  staging:
+    rules:
+      - key: "1"
+        segments: "*"
+        percentage: 100
+
+  production:
+    rules:
+      - key: "1"
+        segments: "*"
+        percentage: 100
+```
+
+## Tagging features
+
+When we create or update any feature in your Featurevisor project, we can tag it with the microfrontend it belongs to:
+
+```yml
+# features/showMarketingBanner.yml
+description: Shows marketing banner at the bottom of the page
+tags:
+  - products # tagging with `products` microfrontend
+
+# ...
+```
+
+It is possible some features may overlap across microfrontends. For example, we may want to show a marketing banner to users in both the `products` and `checkout` microfrontends. In that case, we can tag the feature with both microfrontends:
+
+```yml
+# features/showMarketingBanner.yml
+description: Shows marketing banner at the bottom of the page
+tags:
+  - products
+  - checkout
+
+# ...
+```
+
+## Anonymous vs Authenticated
+
+Featurevisor relies on the feature's `bucketBy` property to determine how to [bucket](/docs/bucketing) the user into a variation.
+
+It's an easy decision to make when choosing the `bucketBy` value when the microfrontend is only accessible to either anonymous or authenticated users, and not both.
+
+| Microfrontend | Access          | `bucketBy` |
+|---------------|-----------------|------------|
+| `signup`      | Anonymous users | `deviceId` |
+| `signin`      | Anonymous users | `deviceId` |
+| `account`     | Authenticated   | `userId`   |
+
+But what shall we do for microfrontends that are accessible to both anonymous and authenticated users? Like `products` and `checkout` in our example.
+
+### Design decision
```

Comment:
This is presented as a Design Decision but in my opinion you're just saying "Option 1 has problem X, you should use Option 2". Is there any specific case when Option 2 has problems and Option 1 shines instead?

Reply:
**TL;DR**: 

- Option 1: use `deviceId` for bucketing at all times
- Option 2: try `userId` for bucketing, if not exists then use `deviceId`

The drawback of Option 2 would be the risk of seeing a feature's variation change if the User's logged in status changes. But it will stay consistent for that logged in User across all sessions/devices.

Option 1 works best if User accesses our app in one device only.

## In other tools

in few other feature flagging tools, we will notice that the concept of a single User ID is part of the API signature in the SDK, like:

```js
sdk.getSomeValue(featureKey, userId, attributes);
```

this allows different client-side applications to choose in the runtime, against which value they wish to bucket a feature in. like against either User ID, Organization ID, Device Id, and so on.

we aren't in control of defining which value to bucket against when designing our feature itself. we rely on client-side applications to take the responsibility of consistently choosing the desired value, which can be error prone.

## In Featurevisor

There's no singular User ID anywhere in the SDK, and everything is just attributes:

```js
sdk.getVariation(featureKey, attributes);
```

and we define how to handle bucketing, that is against which attribute's value the bucketing should take place for the feature, in the feature's YAML itself:

```yml
# features/myFeature.yml
# ...

bucketBy: userId # or deviceId for e.g.

# ...
```

And client-side applications can pass the full attributes object with all possible keys, without having to worry about how bucketing should take place against which attribute:

```js
sdk.getVariation("myFeature", {
  deviceId: "...",
  userId: "...",
});
```

---

This approach moves the responsibility to those who design the feature itself, and they have to make a decision regarding which attribute(e) to bucket against.